src/App.tsx

```diff
@@ -1,61 +1,123 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
 
 export const App = () => {
+  const [titleInput, setTitleInput] = useState('');
+  const [userSelect, setUserSelect] = useState(0);
+  const [todoTextError, setTodoTextError] = useState(false);
+  const [userError, setUserError] = useState(false);
+  const [todos, setTodos] = useState([...todosFromServer]);
+
+  function getMaxId() {
+    const maxId: number = Math.max(...todos.map(todo => todo.id));
+
+    return maxId + 1;
+  }
+
+  function getTodosWithUser(id: number) {
+    return usersFromServer.find(user => user.id === id);
+  }
+
+  const visibleTodos = todos.map(todo => (
+    { ...todo, user: getTodosWithUser(todo.userId) as User }
+  ));
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!titleInput.trim()) {
+      setTodoTextError(true);
+      setTitleInput('');
+    }
+
+    if (userSelect === 0) {
+      setUserError(true);
+    }
+
+    if (userSelect && titleInput.trim()) {
+      setTodos([...todos, {
+        id: getMaxId(),
+        title: titleInput,
+        completed: false,
+        userId: userSelect,
```

Comment:
Can make a function that returns this object. Then call it in this place. ( recommendation)